modules/internal/DefaultRectangular.chpl

```diff
@@ -881,16 +888,43 @@ module DefaultRectangular {
           halt("array index out of bounds: " + _stringify_index(ind));
         }
       var dataInd = getDataIndex(ind);
-      //assert(dataInd >= 0);
-      //assert(numelm >= 0); // ensure it has been initialized
-      //assert(dataInd: uint(64) < numelm: uint(64));
       return theData(dataInd);
     }
-  
+    inline proc dsiAccess(ind : rank*idxType)
+    where isPODType(eltType)
+    {
+      if boundsChecking then
+        if !dom.dsiMember(ind) {
+          halt("array index out of bounds: " + _stringify_index(ind));
+        }
+      var dataInd = getDataIndex(ind);
+      return theData(dataInd);
+    }
+    inline proc dsiAccess(ind : rank*idxType) const ref
+    where !isPODType(eltType)
+    {
+      if boundsChecking then
+        if !dom.dsiMember(ind) {
+          halt("array index out of bounds: " + _stringify_index(ind));
+        }
+      var dataInd = getDataIndex(ind);
+      return theData(dataInd);
+    }
+
     inline proc dsiLocalAccess(i) ref {
       return dsiAccess(i);
     }
-  
+    inline proc dsiLocalAccess(i)
```

Comment:
The dsiLocalAccess variants are unnecessarily bloated in terms of vertical space.
Let's write them like this:

inline proc dsiLocalAccess(i) const ref  where !isPODType(eltType)
  return dsiAccess(i);